firewood/src/merkle/node/partial_path.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+use super::Flags;
+use crate::nibbles::NibblesIterator;
+use std::{
+    fmt::{self, Debug},
+    iter::once,
+};
+
+// TODO: use smallvec
```

Comment:
comment only: We might even get by with Box<[u8]> since we almost always know how big it will be when we start. I'm not sure how often these grow.

Reply:
`Box<[u8]>` would definitely be an improvement, but the goal of `smallvec` or something like it would be quick stack-copies since we actually copy these partial-paths a lot. 

Something like `Bytes` might be more of a benefit, though, where we store the longest keys  on the heap and reference subsets of the keys in the nodes themselves. Again, the goal is a stack-copy when creating a node (branch of leaf) instead of having to allocate copied data.